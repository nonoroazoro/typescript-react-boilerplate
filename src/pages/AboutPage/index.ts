export { AboutPage as default } from "./AboutPage";
export * from "./AboutPage";
