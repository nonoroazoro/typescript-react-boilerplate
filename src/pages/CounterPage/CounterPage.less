@import "../../styles/variables.less";

.container
{
    display: flex;
    justify-content: center;
    align-items: center;

    padding-top: 16px;
}

.innerContainer
{
    background-color: @backgroundColorSecondary;
    border-radius: @borderRadiusMedium;
    box-shadow: @shadowCover;
    padding: 16px 8px;
}

.value
{
    display: inline-block;
    text-align: center;
    min-width: 32px;
}

.btn
{
    .button();

    padding: 8px 16px;
    margin: 0 8px;
    min-width: 60px;

    background-color: @white;
    border-color: @borderColor;
    border-radius: @borderRadiusMedium;

    &:hover
    {
        background-color: @primaryColor;
        border-color: @primaryColor;
    }

    &:active
    {
        background-color: darken(@primaryColor, 8%);
        border-color: darken(@primaryColor, 8%);
    }
}
