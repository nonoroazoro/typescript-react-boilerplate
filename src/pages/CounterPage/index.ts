export { CounterPage as default } from "./CounterPage";
export * from "./CounterPage";
