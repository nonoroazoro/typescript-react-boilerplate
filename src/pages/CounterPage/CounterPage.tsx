import * as cs from "classnames";
import * as React from "react";

import { BaseReactProps } from "../../types";
import { ErrorBoundaryContext } from "../../components/ErrorBoundary";
import { ExampleModal } from "../../components/ExampleModal";
import { queryValue } from "../../api";

import * as styles from "./CounterPage.less";

export interface CounterPageProps extends BaseReactProps
{
    value?: number;
}

export const CounterPage = (props: CounterPageProps) =>
{
    const { className, value: initialValue = 0 } = props;
    const { throwError } = React.useContext(ErrorBoundaryContext);

    const [value, setValue] = React.useState(initialValue);
    const [showModal, setShowModal] = React.useState(false);

    React.useEffect(() =>
    {
        queryValue().then(setValue).catch(throwError);
    }, [throwError]);

    return (
        <div className={cs(className, styles.container)}>
            <div className={styles.innerContainer}>
                <span className={styles.value}>{value}</span>
                <button className={styles.btn} type="button" onClick={_handleClick(1)}>+1</button>
                <button className={styles.btn} type="button" onClick={_handleClick(-1)}>-1</button>
                <button className={styles.btn} type="button" onClick={_handleClick(2)}>+2</button>
                <button className={styles.btn} type="button" onClick={_handleClick(-3)}>-3</button>
                <button className={styles.btn} type="button" onClick={_handleShowModal}>I need a dialog...</button>
            </div>
            <ExampleModal
                show={showModal}
                onOK={_handleHideModal}
                onCancel={_handleHideModal}
            />
        </div>
    );

    function _handleClick(increment: number)
    {
        return () => { setValue((prev) => prev + increment); };
    }

    function _handleShowModal()
    {
        setShowModal(true);
    }

    function _handleHideModal()
    {
        setShowModal(false);
    }
};
