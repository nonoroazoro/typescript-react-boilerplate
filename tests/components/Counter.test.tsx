import { fireEvent, render, waitFor } from "@testing-library/react";

import { CounterPage } from "../../src/pages/CounterPage";

describe("CounterPage", () =>
{
    const props = { className: "counter-page-test", value: 1 };

    it("should render the page", () =>
    {
        const { container } = render(<CounterPage {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it("should render initial value", () =>
    {
        const { getByTestId } = render(<CounterPage {...props} />);
        expect(getByTestId("counter-value").textContent).toBe(String(props.value));
    });

    it("should change value after the increase button is clicked", async () =>
    {
        const { getByTestId, getByText } = render(<CounterPage {...props} />);
        fireEvent.click(getByText("+1"));
        await waitFor(() =>
        {
            expect(getByTestId("counter-value").textContent).toEqual(String(props.value + 1));
        });
    });
});
