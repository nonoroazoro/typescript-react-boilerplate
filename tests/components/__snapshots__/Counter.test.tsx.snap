// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component render component 1`] = `
<div
  className="counter-test container"
>
  <div
    className="innerContainer"
  >
    <span
      className="value"
    >
      0
    </span>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      -1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      -3
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      I need a dialog...
    </button>
  </div>
  <Component
    onCancel={[Function]}
    onOK={[Function]}
    show={false}
  />
</div>
`;
